schema-bridge.graphql

```diff
@@ -426,11 +758,271 @@ type PoolHourlySnapshot @entity {
   rewardTokenEmissionsUSD: [BigDecimal!]
 }
 
+type PoolRouteSnapshot @entity {
+  " {PoolRoute ID}-{poolSnaphotID} "
+  id: ID!
+
+  " PoolRoute this snapshot belongs to "
+  poolRoute: PoolRoute!
+
+  " Timestamp of this snapshot "
+  timestamp: BigInt!
+
+  " Block number of this snapshot "
+  blockNumber: BigInt!
+
+  " volume received through this route during the time of the snapshot period (hourly/daily) "
+  snapshotVolumeIn: BigInt!
+
+  " total volume received through this route in inputToken "
+  cumulativeVolumeIn: BigInt!
+
+  " volume sent through this route during the time of the snapshot period (hourly/daily) "
+  snapshotVolumeOut: BigInt!
+
+  " total volume sent through this route in inputToken "
+  cumulativeVolumeOut: BigInt!
+
+  " volume received through this route during the time of the snapshot period (hourly/daily) in USD "
+  snapshotVolumeInUSD: BigDecimal!
+
+  " total volume received through this route in USD "
+  cumulativeVolumeInUSD: BigDecimal!
+
+  " volume sent through this route during the time of the snapshot period (hourly/daily) in USD "
+  snapshotVolumeOutUSD: BigDecimal!
+
+  " total volume sent through this route in USD "
+  cumulativeVolumeOutUSD: BigDecimal!
+}
+
+##################################
+##### Transaction-Level Data #####
+##################################
+
+"""
+An event is any user action that occurs in a protocol. Generally, they are Ethereum events
+emitted by a function in the smart contracts, stored in transaction receipts as event logs.
+However, some user actions of interest are function calls that don't emit events. For example,
+the deposit and withdraw functions in Yearn do not emit any events. In our subgraphs, we still
+store them as events, although they are not technically Ethereum events emitted by smart
+contracts.
+"""
+interface Event {
+  " { Event type }-{ Transaction hash }-{ Log index } "
```

Comment:
There is no enum called `EventType`, should probably have that or change the id.

Another question is, why have this identifier instead of unique entities for each event type. That way is the precedent and seems to be more clear on the ingestion side.

Reply:
There is also this: https://medium.com/edge-node-engineering/two-simple-subgraph-performance-improvements-a76c6b3e7eac

I think we should use this going forward in any new schemas as it seems to be better practice. Immutable entities and `Bytes` as the id type

Reply:
Sounds good. Will add that too.
How do bytes show when querying? I’m thinking, are we relying on the value of the ID for something anywhere? For example, liquidity pools in DEXes have their contract address as the ID. When querying, do they render as a string?

`Event type` refers to if it is a transfer, deposit, withdrawal, message. Will clarify it 👍

Reply:
single markets show up as the address, it gets a little funky and unreadable when you concat things onto it.

> are we relying on the value of the ID for something anywhere?

What I did to mediate this is add the days or hours as a field to the snapshot

Reply:
Have done the same thing then